airbyte-config/models/src/main/java/io/airbyte/config/ConfigSchema.java

```diff
@@ -27,45 +27,103 @@
+      sourceConnection -> {
+        return sourceConnection.getSourceId().toString();
+      }),
 
   // destination
-  STANDARD_DESTINATION_DEFINITION("StandardDestinationDefinition.yaml"),
-  DESTINATION_CONNECTION("DestinationConnection.yaml"),
+  STANDARD_DESTINATION_DEFINITION("StandardDestinationDefinition.yaml",
+      StandardDestinationDefinition.class, standardDestinationDefinition -> {
+        return standardDestinationDefinition.getDestinationDefinitionId().toString();
+      }),
+  DESTINATION_CONNECTION("DestinationConnection.yaml", DestinationConnection.class,
+      destinationConnection -> {
+        return destinationConnection.getDestinationId().toString();
+      }),
 
   // sync
-  STANDARD_SYNC("StandardSync.yaml"),
-  STANDARD_SYNC_OPERATION("StandardSyncOperation.yaml"),
-  STANDARD_SYNC_SUMMARY("StandardSyncSummary.yaml"),
+  STANDARD_SYNC("StandardSync.yaml", StandardSync.class, standardSync -> {
+    return standardSync.getConnectionId().toString();
+  }),
+  STANDARD_SYNC_OPERATION("StandardSyncOperation.yaml", StandardSyncOperation.class,
+      standardSyncOperation -> {
+        return standardSyncOperation.getOperationId().toString();
+      }),
+  STANDARD_SYNC_SUMMARY("StandardSyncSummary.yaml", StandardSyncSummary.class,
+      standardSyncSummary -> {
+        throw new RuntimeException("No Id for StandardSyncSummary exists");
+      }),
 
   // worker
-  STANDARD_SYNC_INPUT("StandardSyncInput.yaml"),
-  NORMALIZATION_INPUT("NormalizationInput.yaml"),
-  OPERATOR_DBT_INPUT("OperatorDbtInput.yaml"),
+  STANDARD_SYNC_INPUT("StandardSyncInput.yaml", StandardSyncInput.class,
+      standardSyncInput -> {
+        throw new RuntimeException("No Id for StandardSyncInput exists");
+      }),
+  NORMALIZATION_INPUT("NormalizationInput.yaml", NormalizationInput.class,
+      normalizationInput -> {
+        throw new RuntimeException("No Id for NormalizationInput exists");
+      }),
+  OPERATOR_DBT_INPUT("OperatorDbtInput.yaml", OperatorDbtInput.class,
+      operatorDbtInput -> {
+        throw new RuntimeException("No Id for OperatorDbtInput exists");
+      }),
 
-  STANDARD_SYNC_OUTPUT("StandardSyncOutput.yaml"),
-  REPLICATION_OUTPUT("ReplicationOutput.yaml"),
+  STANDARD_SYNC_OUTPUT("StandardSyncOutput.yaml", StandardSyncOutput.class,
+      standardWorkspace -> {
+        throw new RuntimeException("No Id for StandardSyncOutput exists");
+      }),
+  REPLICATION_OUTPUT("ReplicationOutput.yaml", ReplicationOutput.class,
+      standardWorkspace -> {
+        throw new RuntimeException("No Id for ReplicationOutput exists");
+      }),
 
-  STATE("State.yaml");
+  STATE("State.yaml", State.class, standardWorkspace -> {
+    throw new RuntimeException("No Id for State exists");
+  });
 
   static final Path KNOWN_SCHEMAS_ROOT = JsonSchemas.prepareSchemas("types", ConfigSchema.class);
 
   private final String schemaFilename;
+  private final Class<?> className;
+  private final Function<?, String> extractId;
```

Comment:
why do you use `?` instead of keeping the `T`?

Reply:
@michel-tricot since `ConfigSchema` is an enum, we can not use `T`, thats why I had to use `?`